package/help/legacy/options.txt
*options.txt*	For Vim version 9.1.  Last change: 2025 Aug 16


		  VIM REFERENCE MANUAL	  by Bram Moolenaar


Options							*options*

1. Setting options			|set-option|
2. Automatically setting options	|auto-setting|
3. Options summary			|option-summary|

For an overview of options see quickref.txt |option-list|.

Vim has a number of internal variables and switches which can be set to
achieve special effects.  These options come in three forms:
	boolean		can only be on or off		*boolean* *toggle*
	number		has a numeric value
	string		has a string value

==============================================================================
1. Setting options					*set-option* *E764*

							*:se* *:set*
:se[t][!]		Show all options that differ from their default value.
			When [!] is present every option is on a separate
			line.

:se[t][!] all		Show all but terminal options.
			When [!] is present every option is on a separate
			line.

:se[t] termcap		Show all terminal options.  Note that in the GUI the
			key codes are not shown, because they are generated
			internally and can't be changed.  Changing the terminal
			codes in the GUI is not useful either...
			The options have the form t_AB, see
			|terminal-options|.

:se[t]! termcap		Idem, but don't use multiple columns.

								*E518* *E519*
:se[t] {option}?	Show value of {option}.

:se[t] {option}		Toggle option: set, switch it on.
			Number option: show value.
			String option: show value.

:se[t] no{option}	Toggle option: Reset, switch it off.

							   *:set-!* *:set-inv*
:se[t] {option}!   or
:se[t] inv{option}	Toggle option: Invert value.

				*:set-default* *:set-&* *:set-&vi* *:set-&vim*
:se[t] {option}&	Reset option to its default value.  May depend on the
			current value of 'compatible'.
:se[t] {option}&vi	Reset option to its Vi default value.
:se[t] {option}&vim	Reset option to its Vim default value.

:se[t] all&		Set all options to their default value.  The values of
			these options are not changed:
			  all terminal options, starting with t_
			  'columns'
			  'cryptmethod'
			  'encoding'
			  'key'
			  'lines'
			  'term'
			  'ttymouse'
			  'ttytype'
			Warning: This may have a lot of side effects.

					    *:set-args* *:set=* *E487* *E521*
:se[t] {option}={value}		or
:se[t] {option}:{value}
			Set string or number option to {value}.
			For numeric options the value can be given in decimal,
			hex (preceded with 0x) or octal (preceded with '0' or
			'0o').
			The old value can be inserted by typing 'wildchar' (by
			default this is a <Tab> or CTRL-E if 'compatible' is
			set). Many string options with fixed syntax and names
			also support completing known values.  See
			|cmdline-completion| and |complete-set-option|.
			White space between {option} and '=' is allowed and
			will be ignored.  White space between '=' and {value}
			is not allowed.
			See |option-backslash| for using white space and
			backslashes in {value}.

:se[t] {option}+={value}				*:set+=*
			Add the {value} to a number option, or append the
			{value} to a string option.  When the option is a
			comma-separated list, a comma is added, unless the
			value was empty.
			If the option is a list of flags, superfluous flags
			are removed.  When adding a flag that was already
			present the option value doesn't change.
			Also see |:set-args| above.

:se[t] {option}^={value}				*:set^=*
			Multiply the {value} to a number option, or prepend
			the {value} to a string option.  When the option is a
			comma-separated list, a comma is added, unless the
			value was empty.
			Also see |:set-args| above.

:se[t] {option}-={value}				*:set-=*
			Subtract the {value} from a number option, or remove
			the {value} from a string option, if it is there.
			If the {value} is not found in a string option, there
			is no error or warning.  When the option is a comma
			separated list, a comma is deleted, unless the option
			becomes empty.
			When the option is a list of flags, {value} must be
			exactly as they appear in the option.  Remove flags
			one by one to avoid problems.
			The individual values from a comma separated list or
			list of flags can be inserted by typing 'wildchar'.
			See |complete-set-option|.
			Also see |:set-args| above.

The {option} arguments to ":set" may be repeated.  For example: >
	:set ai nosi sw=3 ts=3
If you make an error in one of the arguments, an error message will be given
and the following arguments will be ignored.

							*:set-verbose*
When 'verbose' is non-zero, displaying an option value will also tell where it
was last set.  Example: >
	:verbose set shiftwidth cindent?
<  shiftwidth=4 ~
	  Last set from modeline line 1 ~
  cindent ~
	  Last set from /usr/local/share/vim/vim60/ftplugin/c.vim line 30 ~
This is only done when specific option values are requested, not for ":verbose
set all" or ":verbose set" without an argument.
When the option was set by hand there is no "Last set" message.
When the option was set while executing a function, user command or
autocommand, the script in which it was defined is reported.
Note that an option may also have been set as a side effect of setting
'compatible'.
A few special texts:
	Last set from modeline line 1 ~
		Option was set in a |modeline|.
	Last set from --cmd argument ~
		Option was set with command line argument |--cmd| or +.
	Last set from -c argument ~
		Option was set with command line argument |-c|, +, |-S| or
		|-q|.
	Last set from environment variable ~
		Option was set from an environment variable, $VIMINIT,
		$GVIMINIT or $EXINIT.
	Last set from error handler ~
		Option was cleared when evaluating it resulted in an error.

{not available when compiled without the |+eval| feature}

							*:set-termcap* *E522*
For {option} the form "t_xx" may be used to set a terminal option.  This will
override the value from the termcap.  You can then use it in a mapping.  If
the "xx" part contains special characters, use the <t_xx> form: >
	:set <t_#4>=^[Ot
This can also be used to translate a special code for a normal key.  For
example, if Alt-b produces <Esc>b, use this: >
	:set <M-b>=^[b
(the ^[ is a real <Esc> here, use CTRL-V <Esc> to enter it)
The advantage over a mapping is that it works in all situations.

You can define any key codes, e.g.: >
	:set t_xy=^[foo;
There is no warning for using a name that isn't recognized.  You can map these
codes as you like: >
	:map <t_xy> something
<								*E846*
When a key code is not set, it's like it does not exist.  Trying to get its
value will result in an error: >
	:set t_kb=
	:set t_kb
<	E846: Key code not set: t_kb ~

The t_xx options cannot be set from a |modeline| or in the |sandbox|, for
security reasons.

The listing from ":set" looks different from Vi.  Long string options are put
at the end of the list.  The number of options is quite large.  The output of
"set all" probably does not fit on the screen, causing Vim to give the
|more-prompt|.

							*option-backslash*
To include white space in a string option value it has to be preceded with a
backslash.  To include a backslash you have to use two.  Effectively this
means that the number of backslashes in an option value is halved (rounded
down).
In options 'path', 'cdpath', and 'tags', spaces have to be preceded with three
backslashes instead for compatibility with version 3.0 where the options can
be separated by either commas or spaces.
Comma-separated options like 'backupdir' and 'tags' will also require commas
to be escaped with two backslashes, whereas this is not needed for
non-comma-separated ones like 'makeprg'.
When setting options using |:let| and |literal-string|, you need to use one
fewer layer of backslash.
A few examples: >
   :set makeprg=make\ file	    results in "make file"
   :let &makeprg='make file'	    (same as above)
   :set makeprg=make\\\ file	    results in "make\ file"
   :set tags=tags\ /usr/tags	    results in "tags" and "/usr/tags"
   :set tags=tags\\\ file	    results in "tags file"
   :let &tags='tags\ file'	    (same as above)

   :set makeprg=make,file	    results in "make,file"
   :set makeprg=make\\,file	    results in "make\,file"
   :set tags=tags,file		    results in "tags" and "file"
   :set tags=tags\\,file	    results in "tags\,file"
   :let &tags='tags\,file'	    (same as above)

The "|" character separates a ":set" command from a following command.  To
include the "|" in the option value, use "\|" instead.  This example sets the
'titlestring' option to "hi|there": >
   :set titlestring=hi\|there
This sets the 'titlestring' option to "hi" and 'iconstring' to "there": >
   :set titlestring=hi|set iconstring=there

Similarly, in legacy script the double quote character starts a comment.  To
include the '"' in the option value, use '\"' instead.  This example sets the
'titlestring' option to 'hi "there"': >
   :set titlestring=hi\ \"there\"

In |Vim9| script it's simpler, comments start with a '#' character, and only
when preceded by white space.  A backslash is needed less often: >
   vim9script
   set titlestring=hi\ "there"
   set titlestring=hi#there#
   set titlestring=hi\ \#there#

For Win32 backslashes in file names are mostly not removed.  More precise: For
options that expect a file name (those where environment variables are
expanded) a backslash before a normal file name character is not removed.  But
a backslash before a special character (space, backslash, comma, etc.) is used
like explained above.
There is one special situation, when the value starts with "\\": >
   :set dir=\\machine\path	    results in "\\machine\path"
   :set dir=\\\\machine\\path	    results in "\\machine\path"
   :set dir=\\path\\file	    results in "\\path\file" (wrong!)
For the first one the start is kept, but for the second one the backslashes
are halved.  This makes sure it works both when you expect backslashes to be
halved and when you expect the backslashes to be kept.  The third gives a
result which is probably not what you want.  Avoid it.

				*add-option-flags* *remove-option-flags*
				*E539* *E550* *E551* *E552*
Some options are a list of flags.  When you want to add a flag to such an
option, without changing the existing ones, you can do it like this: >
   :set guioptions+=a
Remove a flag from an option like this: >
   :set guioptions-=a
This removes the 'a' flag from 'guioptions'.
Note that you should add or remove one flag at a time.  If 'guioptions' has
the value "ab", using "set guioptions-=ba" won't work, because the string "ba"
doesn't appear.

			   *:set_env* *expand-env* *expand-environment-var*
Environment variables in specific string options will be expanded.  If the
environment variable exists the '$' and the following environment variable
name is replaced with its value.  If it does not exist the '$' and the name
are not modified.  Any non-id character (not a letter, digit or '_') may
follow the environment variable name.  That character and what follows is
appended to the value of the environment variable.  Examples: >
   :set term=$TERM.new
   :set path=/usr/$INCLUDE,$HOME/include,.
When adding or removing a string from an option with ":set opt-=val" or ":set
opt+=val" the expansion is done before the adding or removing.


Handling of local options			*local-options*

Note: The following also applies to |global-local| options.

Some of the options only apply to a window or buffer.  Each window or buffer
has its own copy of this option, thus each can have its own value.  This
allows you to set 'list' in one window but not in another.  And set
'shiftwidth' to 3 in one buffer and 4 in another.

The following explains what happens to these local options in specific
situations.  You don't really need to know all of this, since Vim mostly uses
the option values you would expect.  Unfortunately, doing what the user
expects is a bit complicated...

When splitting a window, the local options are copied to the new window.  Thus
right after the split the contents of the two windows look the same.

When editing a new buffer, its local option values must be initialized.  Since
the local options of the current buffer might be specifically for that buffer,
these are not used.  Instead, for each buffer-local option there also is a
global value, which is used for new buffers.  With ":set" both the local and
global value is changed.  With "setlocal" only the local value is changed,
thus this value is not used when editing a new buffer.

When editing a buffer that has been edited before, the options from the window
that was last closed are used again.  If this buffer has been edited in this
window, the values from back then are used.  Otherwise the values from the
last closed window where the buffer was edited last are used.

It's possible to set a local window option specifically for a type of buffer.
When you edit another buffer in the same window, you don't want to keep
using these local window options.  Therefore Vim keeps a global value of the
local window options, which is used when editing another buffer.  Each window
has its own copy of these values.  Thus these are local to the window, but
global to all buffers in the window.  With this you can do: >
	:e one
	:set list
	:e two
Now the 'list' option will also be set in "two", since with the ":set list"
command you have also set the global value. >
	:set nolist
	:e one
	:setlocal list
	:e two
Now the 'list' option is not set, because ":set nolist" resets the global
value, ":setlocal list" only changes the local value and ":e two" gets the
global value.  Note that if you do this next: >
	:e one
You will get back the 'list' value as it was the last time you edited "one".
The options local to a window are remembered for each buffer.  This also
happens when the buffer is not loaded, but they are lost when the buffer is
wiped out |:bwipe|.

Special local window options			*local-noglobal*

The following local window options won't be copied over when new windows are
created, thus they behave slightly differently:

	Option		Reason	~
	'previewwindow'	there can only be a single one
	'scroll'	specific to existing window
	'winfixbuf'	specific to existing window
	'winfixheight'	specific to existing window
	'winfixwidth'	specific to existing window

Special local buffer options

The following local buffer options won't be copied over when new buffers are
created, thus they behave slightly differently:

	Option		Reason	~
	'filetype'	explicitly set by autocommands
	'syntax'	explicitly set by autocommands
	'bufhidden'	denote |special-buffers|
	'buftype'	denote |special-buffers|
	'readonly'	will be detected automatically
	'modified'	will be detected automatically

							*:setl* *:setlocal*
:setl[ocal][!] ...	Like ":set" but set only the value local to the
			current buffer or window.  Not all options have a
			local value.  If the option does not have a local
			value the global value is set.
			With the "all" argument: display local values for all
			local options.
			Without argument: Display local values for all local
			options which are different from the default.
			When displaying a specific local option, show the
			local value.  For a global/local boolean option, when
			the global value is being used, "--" is displayed
			before the option name.
			For a global option the global value is
			shown (but that might change in the future).

:se[t] {option}<	Set the effective value of {option} to its global
			value.
			For string |global-local| options, the local value is
			removed, so that the global value will be used.
			For all other options, the global value is copied to
			the local value.

:setl[ocal] {option}<	Set the effective value of {option} to its global
			value.
			For number and boolean |global-local| options, the
			local value is removed, so that the global value will
			be used.
			For all other options, including string |global-local|
			options, the global value is copied to the local
			value.

Note that the behaviour for |global-local| options is slightly different
between string and number-based options.

							*:setg* *:setglobal*
:setg[lobal][!] ...	Like ":set" but set only the global value for a local
			option without changing the local value.
			When displaying an option, the global value is shown.
			With the "all" argument: display global values for all
			local options.
			Without argument: display global values for all local
			options which are different from the default.

For buffer-local and window-local options:
	Command		 global value	  local value	       condition ~
      :set option=value	     set	      set
 :setlocal option=value	      -		      set
:setglobal option=value	     set	       -
      :set option?	      -		     display	 local value is set
      :set option?	    display	       -	 local value is not set
 :setlocal option?	      -		     display
:setglobal option?	    display	       -


Global options with a local value			*global-local*

Options are global when you mostly use one value for all buffers and windows.
For some global options it's useful to sometimes have a different local value.
You can set the local value with ":setlocal".  That buffer or window will then
use the local value, while other buffers and windows continue using the global
value.

For example, you have two windows, both on C source code.  They use the global
'makeprg' option.  If you do this in one of the two windows: >
	:set makeprg=gmake
then the other window will switch to the same value.  There is no need to set
the 'makeprg' option in the other C source window too.
However, if you start editing a Perl file in a new window, you want to use
another 'makeprg' for it, without changing the value used for the C source
files.  You use this command: >
	:setlocal makeprg=perlmake
You can switch back to using the global value by making the local value empty: >
	:setlocal makeprg=
This only works for a string option.  For a number or boolean option you need
to use the "<" flag, like this: >
	:setlocal autoread<
Note that for non-boolean and non-number options using "<" copies the global
value to the local value, it doesn't switch back to using the global value
(that matters when the global value changes later).  You can also use: >
	:set path<
This will make the local value of 'path' empty, so that the global value is
used.  Thus it does the same as: >
	:setlocal path=
Note: In the future more global options can be made |global-local|.  Using
":setlocal" on a global option might work differently then.

						*option-value-function*
Some options ('completefunc', 'findfunc', 'imactivatefunc', 'imstatusfunc',
'omnifunc', 'operatorfunc', 'quickfixtextfunc', 'tagfunc' and 'thesaurusfunc')
are set to a function name or a function reference or a lambda function.  When
using a lambda it will be converted to the name, e.g. "<lambda>123".
Examples:
>
	set opfunc=MyOpFunc
	set opfunc=function('MyOpFunc')
	set opfunc=funcref('MyOpFunc')
	set opfunc={a\ ->\ MyOpFunc(a)}

Set to a script-local function: >
	set opfunc=s:MyLocalFunc
	set opfunc=<SID>MyLocalFunc
In |Vim9| script the "s:" and "<SID>" can be omitted if the function exists in
the script: >
	set opfunc=MyLocalFunc

Set using a funcref variable: >
	let Fn = function('MyTagFunc')
	let &tagfunc = Fn

Set using a lambda expression: >
	let &tagfunc = {t -> MyTagFunc(t)}

Set using a variable with lambda expression: >
	let L = {a, b, c -> MyTagFunc(a, b , c)}
	let &tagfunc = L

In Vim9 script, in a compiled function, you can use a lambda, but a
closure does not work, because the function will be called without the
context of where it was defined.


Setting the filetype

:setf[iletype] [FALLBACK] {filetype}			*:setf* *:setfiletype*
			Set the 'filetype' option to {filetype}, but only if
			not done yet in a sequence of (nested) autocommands.
			This is short for: >
				:if !did_filetype()
				:  setlocal filetype={filetype}
				:endif
<			This command is used in a filetype.vim file to avoid
			setting the 'filetype' option twice, causing different
			settings and syntax files to be loaded.

			When the optional FALLBACK argument is present, a
			later :setfiletype command will override the
			'filetype'.  This is to be used for filetype
			detections that are just a guess.  |did_filetype()|
			will return false after this command.

				*option-window* *optwin*
:bro[wse] se[t]			*:set-browse* *:browse-set* *:opt* *:options*
:opt[ions]		Open a window for viewing and setting all options.
			Options are grouped by function.
			Offers short help for each option.  Hit <CR> on the
			short help to open a help window with more help for
			the option.
			Modify the value of the option and hit <CR> on the
			"set" line to set the new value.  For window and
			buffer specific options, the last accessed window is
			used to set the option value in, unless this is a help
			window, in which case the window below help window is
			used (skipping the option-window).
			{not available when compiled without the |+eval|
			feature}

								*$HOME*
Using "~" is like using "$HOME", but it is only recognized at the start of an
option and after a space or comma.

On Unix systems "~user" can be used too.  It is replaced by the home directory
of user "user".  Example: >
    :set path=~mool/include,/usr/include,.

On Unix systems the form "${HOME}" can be used too.  The name between {} can
contain non-id characters then.  Note that if you want to use this for the

docs/CHANGELOG.txt

[[[Features auto-enabled]]]

[+] FEAT_QUICKFIX
[+] FEAT_EVAL
[+] FEAT_TABPANEL
[+] FEAT_DIFF
[+] FEAT_FOLDING
[+] FEAT_SYN_HL
[+] FEAT_FIND_ID
[+] FEAT_TERMINAL
[+] FEAT_AUTOSHELLDIR
[+] FEAT_JOB_CHANNEL
[+] FEAT_CLIPBOARD
[+] FEAT_FOLDING
[+] FEAT_LINEBREAK
[+] FEAT_BYTEOFF
[+] HAVE_SANDBOX
[+] FEAT_COMPL_FUNC
[+] FEAT_SESSION
[+] FEAT_PROP_POPUP
[+] FEAT_SIGNS
[+] UNIX
[+] FEAT_DND
[+] FEAT_TERMGUICOLORS
[+] FEAT_SPELL
[+] FEAT_BROWSE_CMD
[+] FEAT_VIMINFO
[+] FEAT_TIMERS
[+] FEAT_RELTIME
[+] FEAT_BEVAL_TERM
[+] FEAT_BEVAL
[+] FEAT_MULTI_LANG
[+] FEAT_STL_OPT
[+] FEAT_SEARCH_EXTRA
[+] FEAT_CON_DIALOG
[+] FEAT_GETTEXT
[+] FEAT_TERMRESPONSE
[+] FEAT_CLIENTSERVER
[+] FEAT_PERSISTENT_UNDO
[+] FEAT_XATTR
[+] FEAT_FILTERPIPE
[+] FEAT_WRITEBACKUP (seems to not affect anything, even in old Vim)
[+] p_wb ("writebackup" option) - make it always on
[+] HAVE_TGETENT
[+] TERMINFO
[+] FEAT_CSCOPE
[+] kitty_keyboard_protocol (CSU u sequences)
[+] FEAT_TERM_POPUP_MENU
[+] FNAME_ILLEGAL
[+] CURSOR_SHAPE
[+] HAVE_SETENV
[+] MESSAGE_QUEUE
[+] HAS_MESSAGE_WINDOW
[+] VIM_BACKTICK


[[[Deleted cruft]]]

[-] cpoptions
[-] p_ro, readonly mode (merged into unmodifiable)
[-] readonlymode, BufLocal.readonly - finish merging them into unmodifiable
[-] FEAT_VARTABS vartabstop (tabs that have different widths in same document)
[-] FEAT_DIGRAPHS digraphs (replace with action in popup menu?)
[-] input methods (XIM, IME)
[-] FEAT_KEYMAP
[-] FEAT_SOUND
[-] FEAT_MENU
[-] BACKSLASH_IN_FILENAME
[-] FEAT_CONCEAL
[-] AMIGA
[-] VMS
[-] MSWIN
[-] FEAT_GUI
[-] VIMDLL
[-] FEAT_TOOLBAR
[-] FEAT_VTP
[-] FEAT_MZSCHEME
[-] FEAT_BROWSE
[-] FEAT_WAK
[-] beeping (misc.c) 
[-] FEAT_AUTOCHDIR
[-] FEAT_BEVAL_TIP
[-] 'compatible'
[-] FEAT_MOUSESHAPE
[-] FEAT_RENDER_OPTIONS
[-] FEAT_PROFILE
[-] -C, -N flags
[-] encodings other than UTF-8
[-] USE_ICONV, DYNAMIC_ICONV
[-] USE_SYSTEM
[-] pastetoggle
[-] paste mode
[-] tearoff
[-] popups in mousemodel, and the mousemodel option itself
[-] FEAT_PRINTER, FEAT_POSTSCRIPT, hardcopy
[-] FEAT_SIGN_ICONS
[-] SUN_SYSTEM
[-] ELAPSED_FUNC
[-] p_write
[-] Mac OS aka the Walled Garden OS
[-] Select mode
[-] octal in var.c
[-] Ex modes
[-] dlopen (for improvement of security - no runtime code loading)
[-] Select mode
[-] Replace mode
[-] Visual Replace character pending mode
[-] :cgetfile, :cgetbuffer, :lgetfile, :lgetbuffer
[-] :cdo, :cfdo (use :ldo, :lfdo instead)
[-] "cedit" option (just have Ctrl-F as the regular mapping for opening command portal)
[-] "insertmode" option, p_im (silly thing where Vim pretends to be "modeless")
[-] "more files to edit" error
[-] vim9script
[-] p_cpm (clipmethod)
[-] superfluous text decorations like HL_STANDOUT, HL_UNDERDOUBLE
[-] "magic" option (p_magic), :smagic and :snomagic commands
[-] joinspaces option (p_js) (two spaces after sentence ends when J)
[-] backspace option (p_bs)
[-] belloff option (p_bo)
[-] bioskey option
[-] macatsui option
[-] conskey option
[-] termwintype option (p_twt)
[-] guipty option
[-] guitablabel option
[-] guitabtooltip option
[-] hkmap, hkmapp options
[-] p_hl (highlight option)
[-] modelines (b_p_ml, modelines (p_mls), modelineexpr (p_mle)
[-] ldo, lgrep, lmake etc
[-] MCH_UPDATE_CURSOR (some Windows thing)
[-] USE_ON_FLY_SCROLL (some GUI thing)
[-] rightleft, rightleftcmd options (right-to-left viewing)
[-] Solaris mentions
[-] ACL (Git will provide access control, no need to do it at file level)
[-] tabline (tabpanel is enough)
[-] p_tbi (always use system terminfo)
[-] cygwin
[-] class_T, cctx_T (remnants of Vim9script)
[-] p_ff, p_ffs "file formats" (different line endings between Unix and Dos)
[-] p_pi (preserveindent)
[-] p_ci (copyindent)
[-] encoding conversions (p_enc, VimConv, vc_type ...)
[-] p_exrc (loading scripts from some extra place)
[-] selectmode option (p_slm)
[-] p_rtp (runtimepath) and p_pp (packpath)
[-] scriptversion
[-] p_wmw, p_wmh (fixed minimal portal width and height at 10)
[-] clipmethod, clipreset
[-] p_icon, p_iconstring
[-] p_vb (visual bell)
[-] crypto-stuff, DYNAMIC_SODIUM (not the Unix way, encryption is an orthogonal concern)
[-] virtualedit (baked in the "block" value for block visual mode)
[-] p_eb (ring bell on errors)
[-] p_fixeol, p_eol (auto end of line at file end)
[-] p_fdn (foldnestmax, fixed at 32)
[-] foldcolumn (weird markers for nested folds on the left)
[-] p_ambw (ambiwidth) and p_emoji
[-] p_ed (edcompatible option)
[-] p_gd (gdefault - now "g" is on by default in substitutions)
[-] p_sr (shift round - indent with "<" and ">" always rounds to multiples of shiftwidth)
[-] p_smc (max col for syntax coloring, fixed at 256)
[-] p_cdh (cd home, no effect on Unix)
[-] p_wiv (weirdinvert, some legacy thing)
[-] p_sc (showcmd, fixed at "true", this just means commands being typed are shown at bottom line)
[-] p_cb (clipboard)
[-] p_sn (short, MSDOS-style file names)
[-] p_sta (smarttab), p_ts (tab stop), p_sts (softtabstop)
[-] window title & icon
[-] restricted, check_restricted()
[-] suspending, c_stop (useless)
[-] p_bomb (Byte-order mark, evil)
[-] p_dir (directory, location for swap files). Fixed it at ~/.local/state/eegl
[-] p_mat (matchtime, ui delay in long searches)
[-] p_hf (name of main help file, hardcoded)
[-] p_section, p_para (nroff macro support)
[-] p_window (some weird Vi-compatibility thing)
[ ] p_more? (weird Vi-compatible crap) (but there's msg printing code in channel.c that exploits it)
[-] bufhidden (cruft that could close a buffer if there are no portals into it)
[-] weird mouse protocols (FEAT_MOUSE_XTERM, FEAT_MOUSE_JSB etc)
[-] manual foldmethod
[-] syntax foldmethod
[-] p_xtermcodes
[-] p_lisp, p_lop, p_lispwords (lisp)
[-] p_pvp (preview popups, always on)
[-] p_ek (@esckeys, always on)
[-] p_tpm (max number of tabs)
[-] p_to (tilde as operator)
[-] p_ws (searches wrap around file end)
[-] p_uc (writing to swap files after this many chars)
[-] p_qe (quoteescape character, it's always the backslash)
[-] p_fml (min foldable number of lines, useless)

src/indices/options.h
//Automatically generated code by the `make indices` command.
typedef struct {CS name; Unt index;} NameIndex;
static NameIndex const NAME_INDICES[194] = {
{(Byte*)"autocomplete", 0},
{(Byte*)"autocompletedelay", 1},
{(Byte*)"autoindent", 131072},
{(Byte*)"autoread", 131113},
{(Byte*)"autoshelldir", 2},
{(Byte*)"autowrite", 3},
{(Byte*)"autowriteall", 4},
{(Byte*)"backup", 5},
{(Byte*)"backupcopy", 131073},
{(Byte*)"backupdir", 6},
{(Byte*)"backupext", 7},
{(Byte*)"backupskip", 8},
{(Byte*)"balloondelay", 9},
{(Byte*)"balloonevalterm", 10},
{(Byte*)"balloonexpr", 131074},
{(Byte*)"binary", 131075},
{(Byte*)"booklisted", 131076},
{(Byte*)"booktype", 131077},
{(Byte*)"breakindent", 65536},
{(Byte*)"breakindentopt", 65537},
{(Byte*)"cdpath", 11},
{(Byte*)"cinwords", 131078},
{(Byte*)"columns", 14},
{(Byte*)"comments", 131079},
{(Byte*)"commentstring", 131080},
{(Byte*)"commheight", 12},
{(Byte*)"commportheight", 13},
{(Byte*)"complete", 131081},
{(Byte*)"completefunc", 131083},
{(Byte*)"completefuzzycollect", 15},
{(Byte*)"completeitemalign", 16},
{(Byte*)"completeopt", 131082},
{(Byte*)"completepopup", 17},
{(Byte*)"confirm", 18},
{(Byte*)"cscopepathcomp", 19},
{(Byte*)"cscopeprog", 20},
{(Byte*)"cscoperelative", 21},
{(Byte*)"cscopetag", 22},
{(Byte*)"cscopetagorder", 23},
{(Byte*)"cscopeverbose", 24},
{(Byte*)"cursorInsert", 25},
{(Byte*)"cursorNormal", 26},
{(Byte*)"cursorline", 65554},
{(Byte*)"debug", 27},
{(Byte*)"define", 131093},
{(Byte*)"delcombine", 28},
{(Byte*)"dictionary", 131116},
{(Byte*)"diff", 65538},
{(Byte*)"diffanchors", 131117},
{(Byte*)"diffexpr", 29},
{(Byte*)"diffopt", 30},
{(Byte*)"eadirection", 31},
{(Byte*)"eeglinfo", 32},
{(Byte*)"eeglinfofile", 33},
{(Byte*)"equalalways", 34},
{(Byte*)"errorfile", 35},
{(Byte*)"errorformat", 131112},
{(Byte*)"eventignore", 36},
{(Byte*)"eventignoreport", 65539},
{(Byte*)"expandtab", 131087},
{(Byte*)"fillchars", 112},
{(Byte*)"findfunc", 131086},
{(Byte*)"foldenable", 65540},
{(Byte*)"foldexpr", 65544},
{(Byte*)"foldignore", 65541},
{(Byte*)"foldlevel", 65542},
{(Byte*)"foldlevelstart", 37},
{(Byte*)"foldmarker", 65546},
{(Byte*)"foldmethod", 65543},
{(Byte*)"foldopen", 38},
{(Byte*)"foldtext", 65545},
{(Byte*)"formatexpr", 131098},
{(Byte*)"formatlistpat", 131089},
{(Byte*)"formatoptions", 131088},
{(Byte*)"formatprog", 131097},
{(Byte*)"fsync", 39},
{(Byte*)"grepformat", 131108},
{(Byte*)"grepprog", 131109},
{(Byte*)"helpheight", 40},
{(Byte*)"helplang", 41},
{(Byte*)"history", 42},
{(Byte*)"hlsearch", 43},
{(Byte*)"ignorecase", 44},
{(Byte*)"include", 131094},
{(Byte*)"includeexpr", 131095},
{(Byte*)"incsearch", 45},
{(Byte*)"indentexpr", 131096},
{(Byte*)"infercase", 131090},
{(Byte*)"intro", 46},
{(Byte*)"isexpand", 131091},
{(Byte*)"isfname", 47},
{(Byte*)"isident", 48},
{(Byte*)"iskeyword", 131092},
{(Byte*)"keywordprog", 131099},
{(Byte*)"langmap", 49},
{(Byte*)"langremap", 50},
{(Byte*)"lazyredraw", 51},
{(Byte*)"lines", 52},
{(Byte*)"list", 65547},
{(Byte*)"listchars", 111},
{(Byte*)"lowPrioSuffixes", 53},
{(Byte*)"makeOpenWhenDone", 55},
{(Byte*)"makeef", 54},
{(Byte*)"makeprog", 131111},
{(Byte*)"matchpairs", 131100},
{(Byte*)"maxfuncdepth", 56},
{(Byte*)"maxmem", 57},
{(Byte*)"maxmempattern", 58},
{(Byte*)"maxsearchcount", 59},
{(Byte*)"messagesopt", 60},
{(Byte*)"modifiable", 131101},
{(Byte*)"more", 61},
{(Byte*)"mousetime", 62},
{(Byte*)"numberwidth", 65549},
{(Byte*)"omnifunc", 131084},
{(Byte*)"operatorfunc", 63},
{(Byte*)"patchexpr", 64},
{(Byte*)"path", 131110},
{(Byte*)"portfixbuf", 65550},
{(Byte*)"portfixheight", 65551},
{(Byte*)"portfixwidth", 65552},
{(Byte*)"previewheight", 65},
{(Byte*)"pumheight", 66},
{(Byte*)"pummaxwidth", 67},
{(Byte*)"pumwidth", 68},
{(Byte*)"quickfixtextfunc", 69},
{(Byte*)"redrawtime", 70},
{(Byte*)"relativenumber", 65548},
{(Byte*)"rulerformat", 71},
{(Byte*)"scrolljump", 72},
{(Byte*)"scrolloff", 65559},
{(Byte*)"scrollopt", 73},
{(Byte*)"shiftwidth", 131102},
{(Byte*)"showbreak", 113},
{(Byte*)"showcmdloc", 74},
{(Byte*)"showfulltag", 75},
{(Byte*)"showmode", 76},
{(Byte*)"showtabpanel", 77},
{(Byte*)"sidescroll", 78},
{(Byte*)"sidescrolloff", 65558},
{(Byte*)"signcolumn", 65557},
{(Byte*)"smartcase", 79},
{(Byte*)"smartindent", 131104},
{(Byte*)"smoothscroll", 65553},
{(Byte*)"splitbelow", 80},
{(Byte*)"splitright", 81},
{(Byte*)"startofline", 82},
{(Byte*)"statusline", 65555},
{(Byte*)"suffixesadd", 131105},
{(Byte*)"swapfile", 131103},
{(Byte*)"swapsync", 83},
{(Byte*)"switchbook", 84},
{(Byte*)"tabclose", 85},
{(Byte*)"tabpanel", 86},
{(Byte*)"tabpanelopt", 87},
{(Byte*)"tagbsearch", 88},
{(Byte*)"tagcase", 131115},
{(Byte*)"tagfunc", 131085},
{(Byte*)"tags", 131114},
{(Byte*)"tagstack", 89},
{(Byte*)"term", 90},
{(Byte*)"termwinkey", 65560},
{(Byte*)"termwinscroll", 115},
{(Byte*)"termwinsize", 65561},
{(Byte*)"textwidth", 131106},
{(Byte*)"thesaurus", 131118},
{(Byte*)"thesaurusfunc", 131119},
{(Byte*)"timeout", 91},
{(Byte*)"timeoutlen", 92},
{(Byte*)"ttimeout", 93},
{(Byte*)"ttimeoutlen", 94},
{(Byte*)"ttyscroll", 95},
{(Byte*)"undofile", 131120},
{(Byte*)"undolevels", 114},
{(Byte*)"undoreload", 96},
{(Byte*)"updatetime", 97},
{(Byte*)"verbose", 98},
{(Byte*)"verbosefile", 99},
{(Byte*)"whichwrap", 100},
{(Byte*)"wildchar", 101},
{(Byte*)"wildcharm", 102},
{(Byte*)"wildignore", 103},
{(Byte*)"wildignorecase", 104},
{(Byte*)"wildmenu", 105},
{(Byte*)"wildmode", 106},
{(Byte*)"wildoptions", 107},
{(Byte*)"winheight", 108},
{(Byte*)"winwidth", 109},
{(Byte*)"wlseat", 116},
{(Byte*)"wlsteal", 117},
{(Byte*)"wltimeoutlen", 118},
{(Byte*)"wrap", 65556},
{(Byte*)"wrapmargin", 131107},
{(Byte*)"writedelay", 110},
};

static int const FIRST_LETTER_INDICES[27] = {
0,
7,
20,
43,
51,
60,
76,
78,
82,
93,
93,
94,
101,
113,
114,
116,
125,
126,
129,
152,
172,
176,
178,
194,
194,
194,
194,
};

#define OPTION_COUNT 194
